### PR TITLE
fix(files): restore detail actions and preserve rename extensions

### LINKED ISSUE #358: bug(files): Open, Download, and Rename actions disabled — need WebView-compatible approach
## Summary

Three file actions in the Files overlay detail view are broken and have been disabled in the UI:

- **Open ↗** — `window.open(URL.createObjectURL(blob))` is blocked in the Office WebView sandbox
- **Download** — `workspace.downloadFile()` fails silently; the WebView doesn't support programmatic downloads
- **Rename** — works, but if the user omits the file extension the file loses it silently (e.g. `report.xlsx` → `report`). Needs extension preservation or at least a warning.

These were removed from the detail view in the UI cleanup PR. Only **Delete** is exposed for now (it works correctly).

## To re-enable

1. **Open**: Investigate `Office.context.ui.displayDialogAsync` or a helper-page approach as an alternative to `window.open`
2. **Download**: Same constraint — may need to base64-encode and post to a dialog, or use the proxy server as a download relay
3. **Rename**: Preserve extension when user input doesn't include one (e.g. user types `report` for `report.xlsx` → resolve to `report.xlsx`). Consider showing the extension as a fixed suffix in the rename dialog.

## Related

- Supersedes #316 (closed — original bug report)
- #272 (holistic overlay simplification)

### PR DESCRIPTION
**Pull request by tmustier**
Closes #358

## Summary
- restore file detail actions in the Files overlay (`Open ↗`, `Download`, `Rename`, `Delete`) with read-only handling (`Open`/`Download` only)
- harden `Open ↗` by opening a pending tab synchronously before async file reads (reduces popup blocking in Office WebView)
- preserve file extensions during rename when users omit one (`report.xlsx` + `report-final` → `report-final.xlsx`), including path-only and folder-move inputs
- add focused path-resolution helpers and unit coverage for rename edge cases
- document the updated Files detail-action behavior in `src/tools/DECISIONS.md`

## Testing
- `npm run check`
- `npm run build`
- `npm run test:models`
- `npm run test:context`

### CODE REVIEW

File: src/ui/files-dialog.ts [hunk 1]
```diff
@@ -552,19 +599,136 @@ export async function showFilesWorkspaceDialog(): Promise<void> {
...
+          locationKind: fileRef.locationKind,
+        });
+
+        if (result.text === undefined || result.truncated) {
+          throw new Error("File is too large to open in a browser tab.");
+        }
+
+        const blob = new Blob([result.text], {
+          type: file.mimeType || "text/plain",
+        });
```
[the remark this answers is not in the document]
Comment (tmustier) [reply src/ui/files-dialog.ts:622]: Good catch — fixed in 5b5ffdf. `Open ↗` now sanitizes script-capable MIME types via `resolveSafeFilesDialogBlobMimeType(...)`, matching the active-content protection used by download paths. Added tests in `tests/files-dialog-filtering.test.ts` for downgraded MIME cases.
